ha/core/controllers/pcs/service_controller.py

```diff
@@ -52,20 +56,34 @@ def start(self, service: str, nodeids: list = None) -> dict:
         raise HAUnimplemented("This operation is not implemented.")
 
     @controller_error_handler
-    def stop(self, service: str, nodeids: list = None) -> dict:
+    def stop(self, node_id: str, excludeResourceList: list = None) -> dict:
         """
         Stop service.
 
         Args:
-            service (str): Service name.
-            nodeids (list, optional): Node ids, if none then all node status.
-                    Defaults to None.
+            node_id (str): Private fqdn define in conf store.
+            excludeResourceList (list): Service list which are not stopped.
 
         Returns:
             ([dict]): Return dictionary. {"status": "", "output":"", "error"}
                 status: Succeeded, Failed, InProgress
         """
-        raise HAUnimplemented("This operation is not implemented.")
+        try:
+            resources: list = []
+            output, _, _ = self._execute.run_cmd(const.LIST_PCS_RESOURCES, check_error=False)
+            if "NO resources".lower() not in output.lower():
+                for resource in output.split("\n"):
+                    res = resource.split(":")[0]
+                    if res != "" and res not in resources and res not in excludeResourceList:
+                        resources.append(res)
+            for resource in resources:
+                self._execute.run_cmd(f"pcs resource ban {resource} {node_id}")
```

Comment:
Use a define from ha/const.py instead of pcs command directly

Reply:
Done